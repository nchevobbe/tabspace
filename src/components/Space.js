// @flow
import type { SpaceIdType } from "../types.flow";
import React from "react";

type PropsType = {
  active: boolean,
  id: SpaceIdType,
  title: string,
  deleteSpace?: SpaceIdType => void,
  selectSpace?: SpaceIdType => void,
};

export default (props: PropsType) => (
  <li
    className={props.active ? "active" : null}
    onClick={() => {
      if (props.selectSpace) {
        props.selectSpace(props.id);
      }
    }}
  >
    <span>{props.title}</span>
    {props.deleteSpace ? (
      <button
        className="space-delete clean"
        title="Delete space"
        onClick={e => {
          if (props.deleteSpace) {
            props.deleteSpace(props.id);
          }
          e.stopPropagation();
        }}
      />
    ) : null}
  </li>
);
