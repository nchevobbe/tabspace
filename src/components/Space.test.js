import React from "react";
import { shallow } from "enzyme";

import Space from "./Space";

describe("Space", () => {
  it("renders active space as expected", () => {
    expect(shallow(<Space active={true} id="1" title="test" />)).toMatchSnapshot();
  });

  it("renders inactive space as expected", () => {
    expect(shallow(<Space active={false} id="1" title="test" />)).toMatchSnapshot();
  });

  it("handles selectSpace prop as expected", () => {
    const selectSpace = jest.fn();
    const id = Symbol();
    const element = shallow(<Space id={id} title="test" selectSpace={selectSpace} />);
    expect(element).toMatchSnapshot();

    element.simulate("click", { type: "click" });
    expect(selectSpace.mock.calls.length).toEqual(1);
    expect(selectSpace.mock.calls[0][0]).toEqual(id);
  });

  it("handles deleteSpace prop as expected", () => {
    const deleteSpace = jest.fn();
    const stopPropagation = jest.fn();
    const id = Symbol();
    const element = shallow(<Space id={id} title="test" deleteSpace={deleteSpace} />);
    expect(element).toMatchSnapshot();

    const node = element.find(".space-delete");
    node.simulate("click", { type: "click", stopPropagation });

    expect(node.exists()).toBeTruthy();
    expect(deleteSpace.mock.calls.length).toEqual(1);
    expect(deleteSpace.mock.calls[0][0]).toEqual(id);

    expect(stopPropagation.mock.calls.length).toEqual(1);
  });
});
