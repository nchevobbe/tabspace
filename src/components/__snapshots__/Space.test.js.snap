// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space handles deleteSpace prop as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    deleteSpace={[Function]}
    id={Symbol()}
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          test
        </span>,
        <button
          className="space-delete clean"
          onClick={[Function]}
          title="Delete space"
        />,
      ],
      "className": null,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "space-delete clean",
          "onClick": [Function],
          "title": "Delete space",
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            test
          </span>,
          <button
            className="space-delete clean"
            onClick={[Function]}
            title="Delete space"
          />,
        ],
        "className": null,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "space-delete clean",
            "onClick": [Function],
            "title": "Delete space",
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Space handles selectSpace prop as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    id={Symbol()}
    selectSpace={[Function]}
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          test
        </span>,
        null,
      ],
      "className": null,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "span",
      },
      null,
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            test
          </span>,
          null,
        ],
        "className": null,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "span",
        },
        null,
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Space renders active space as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    active={true}
    id="1"
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          test
        </span>,
        null,
      ],
      "className": "active",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "span",
      },
      null,
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            test
          </span>,
          null,
        ],
        "className": "active",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "span",
        },
        null,
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Space renders inactive space as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    active={false}
    id="1"
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          test
        </span>,
        null,
      ],
      "className": null,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "span",
      },
      null,
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            test
          </span>,
          null,
        ],
        "className": null,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "span",
        },
        null,
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
